reframe/core/schedulers/slurm.py

```diff
@@ -267,6 +267,12 @@ def _check_and_cancel(self, reason_descr):
             reason, reason_details = reason_descr, None
 
         if reason in self._cancel_reasons:
+            # Here we handle the case were the UnavailableNodes list is empty,
+            # which actually means that the job is pending
+            if reason == 'ReqNodeNotAvail' and reason_details:
+                if not re.match(r'UnavailableNodes:$', reason_details):
```

Comment:
The `not` is wrong here. This regex matches only when there is nothing after the `:`.